src/main/resources/services/pathstats/pathstats.js

```diff
@@ -0,0 +1,111 @@
+const mustache = require('/lib/mustache');
+const portal = require('/lib/xp/portal');
+const nodeLib  = require('/lib/xp/node');
+
+function parseQueryString(queryString) {
+  if (!queryString) {
+    return null;
+  }
+
+  return queryString.split('&').reduce((acc, pair) => {
+    const parts = pair.split('=');
+    const key = parts[0];
+    const value = parts[1];
```

Comment:
may fail for legal `?q`

Reply:
I tested with a querystring like `?q&r=&s` and in that case all three params are counted, and I've also tested that the value is counted as `null` no matter if the param key is solo like `?q` or with empty value like `?q=`

There might be other edge cases that I don't know about, though.